core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -857,8 +857,13 @@ private[spark] object ApiHelper {
   }
 
   def lastStageNameAndDescription(store: AppStatusStore, job: JobData): (String, String) = {
-    val stage = store.asOption(store.stageAttempt(job.stageIds.max, 0)._1)
-    (stage.map(_.name).getOrElse(""), stage.flatMap(_.description).getOrElse(job.name))
+    // Some jobs have only 0 partition.
+    if (job.stageIds.isEmpty) {
+      ("", "This job has 0 partition")
```

Comment:
partition -> partitions
Does the job not have a name in this case? maybe use it or else "" rather than a somewhat arbitrary description as "a job with 0 partitions"

Reply:
Er....You are right. It's really not appropriate. It's last stage name in a job not a job name. But in this case, We don't have any stage. I think the most suitable name is the `callsite` of running job.

Reply:
So, I think adding a parameter in `SparkListenerJobStart` may be ok. I can't think other method in it.

Reply:
Or constructing a empty stage contains a callsite in StageInfo?

Reply:
No, I wouldn't add a parameter. Just use job.name if available like the current code does.

Reply:
Thanks! fixed.